saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/sync/SavedSitesSyncDataProvider.kt

```diff
@@ -96,79 +100,53 @@ class SavedSitesSyncDataProvider @Inject constructor(
             return emptyList()
         }
 
-        val updates = mutableListOf<SyncBookmarkEntry>()
+        val updates = mutableListOf<SyncSavedSitesRequestEntry>()
         // favorites (we don't add individual items, they are added as we go through bookmark folders)
         if (hasFavorites) {
             val favoritesFolders = listOf(SavedSitesNames.FAVORITES_ROOT, SavedSitesNames.FAVORITES_MOBILE_ROOT)
             favoritesFolders.forEach {
-                val favorites = syncSavedSitesRepository.getFavoritesSync(it)
                 val favoriteFolder = repository.getFolder(it)
                 favoriteFolder?.let {
-                    updates.add(encryptFolder(favoriteFolder, favorites.map { it.id }))
+                    updates.add(encryptedFolder(favoriteFolder))
                 }
             }
         }
 
-        addFolderEntryWithContent(SavedSitesNames.BOOKMARKS_ROOT, updates)
+        addAllFolders(SavedSitesNames.BOOKMARKS_ROOT, updates)
         return updates.distinct()
     }
 
-    private fun addFolderEntry(folderId: String): SyncBookmarkEntry? {
-        val folderContent = repository.getAllFolderContentSync(folderId)
-        val storedFolder = repository.getFolder(folderId)
-        val folder = if (storedFolder != null) {
-            val childrenIds = mutableListOf<String>()
-            for (bookmark in folderContent.first) {
-                if (bookmark.deleted == null) {
-                    childrenIds.add(bookmark.id)
-                }
-            }
-            for (eachFolder in folderContent.second) {
-                if (eachFolder.deleted == null) {
-                    childrenIds.add(eachFolder.id)
-                }
-            }
-            encryptFolder(storedFolder, childrenIds)
-        } else {
-            null
-        }
-        return folder
-    }
-
-    private fun addFolderEntryWithContent(
+    private fun addAllFolders(
         folderId: String,
-        updates: MutableList<SyncBookmarkEntry>,
-    ): List<SyncBookmarkEntry> {
+        updates: MutableList<SyncSavedSitesRequestEntry>,
+    ): List<SyncSavedSitesRequestEntry> {
         repository.getAllFolderContentSync(folderId).apply {
             val folder = repository.getFolder(folderId)
```

Comment:
There a few methods we can move into sync, but other would require to move the favourites delegate and other refactoring that I’d prefer not doing now. 

Regarding the models, I don’t want to change them now since the new Bookmarks UI is already changing the UI models. BookmarkFolder will become a domain model, that we can then modify if needed. Besides being nicer to read, there’s not really any improvement to do it now.